src/workers/scraper.ts

```diff
@@ -1,32 +1,34 @@
 import { Job } from 'bullmq';
 
-import config from '@/config/config';
 import { SearchResultStatus } from '@/constants/enum';
 import { updateSearchResult } from '@/repositories/searchResult';
-import Parser from '@/services/parser';
-import { searchResult } from '@/services/searchResult';
-
-const { selectorIdsSearch } = config;
+import Parser from '@/services/parser/parser';
+import { searchResult } from '@/services/search/searchResult';
 
 export const scrape = async (job: Job) => {
-  const { keyword, userId, id } = job.data;
+  const { keyword, userId, id, searchEngine } = job.data;
   try {
-    const htmlData = await searchResult(keyword);
-    const parserInstance = new Parser(selectorIdsSearch, htmlData);
-    const topAdParsed = parserInstance.parseTopAdWordList();
-    const bottomAdParsed = parserInstance.parseBodyAdWordList();
-    const nonAdParsed = parserInstance.parseNonAdWordList();
-    const adWordsCount = topAdParsed.count + bottomAdParsed.count;
-    const nonAdUrls = nonAdParsed.urls;
+    const htmlData = await searchResult(keyword, searchEngine);
+    const parserInstance = new Parser(searchEngine, htmlData).getParserInstance();
+    const topAdUrls = parserInstance.parseTopAdWordList();
+    const bottomAdUrls = parserInstance.parseBottomAdWordList();
+    const nonAdUrls = parserInstance.parseNonAdWordList();
+    const adWordsCount = topAdUrls.length + bottomAdUrls.length;
+    const totalLinksCount = adWordsCount + nonAdUrls.length;
```

Comment:
A line break would be nice for readability

```suggestion
    const totalLinksCount = adWordsCount + nonAdUrls.length;
    
```